### PR TITLE
Upgrading to Graylog 5

### PR DESCRIPTION
**Pull request by luizjr**
I made the necessary changes to switch from Elasticserach to Opensearch, already using Mongo 6 with Graylog 5.

I'm starting from scratch, I still don't know the impact it can have on applications that are already online... because it's a logging tool, I programmed the update and simply killed the old one and started this update from scratch and it worked fine ...

### CODE REVIEW

**Comment by KongZ** [comment]
@luizjr Thanks for greatly contributing. We need to keep chart sources reference to my repository but you can add your name as another author/maintainers.

File: charts/graylog/values.yaml [hunk 1]
```diff
@@ -55,11 +55,12 @@ graylog:
...
   image:
     repository: "graylog/graylog"
-    tag: "4.2.3-1"
+    tag: "5.0.2"
     pullPolicy: "IfNotPresent"
 
   ## Graylog default Java option
-  javaOpts: "-Dlog4j2.formatMsgNoLookups=true -Djdk.tls.acknowledgeCloseNotify=true -XX:+UnlockExperimentalVMOptions -XX:NewRatio=1 -XX:MaxMetaspaceSize=256m -server -XX:+ResizeTLAB -XX:-OmitStackTraceInFastThrow"
+  # javaOpts: "-Dlog4j2.formatMsgNoLookups=true -Djdk.tls.acknowledgeCloseNotify=true -XX:+UnlockExperimentalVMOptions -XX:-OmitStackTraceInFastThrow -XX:+UseG1GC -server"
+  javaOpts: "-Dlog4j2.formatMsgNoLookups=true -Djdk.tls.acknowledgeCloseNotify=true -XX:+UnlockExperimentalVMOptions -XX:-OmitStackTraceInFastThrow -XX:+UseG1GC -server"
```
Comment (pauronda-visma) [inline charts/graylog/values.yaml:63]: I had problems with the following parameter:
-XX:+UseCGroupMemoryLimitForHeap
It is being set on line [109 charts/graylog/templates/statefulset.yaml](https://github.com/KongZ/charts/blob/main/charts/graylog/templates/statefulset.yaml#L109)
After removing it worked for me.
Be sure it works.
More info in the graylog repo:
[Link to Graylog Repo](https://github.com/Graylog2/graylog-docker/blob/5.0/docker/oss/Dockerfile#L67)

**Review by pauronda-visma — COMMENTED** [review]
Please be sure that -XX:+UseCGroupMemoryLimitForHeap doesn't affect

**Comment by luizjr** [comment]
> @luizjr Thanks for greatly contributing. We need to keep chart sources reference to my repository but you can add your name as another author/maintainers.

The intention is not to duplicate, I just uploaded it that way, because I had to apply the update faster.
I can fix that if you update the chart, so I can use it instead of the one I made temporarily...
I'm just worried about whoever upgrades and can't recreate everything like I did...
I still don't know how the data migration from elastic to opensearch will be

**Comment by KongZ** [comment]
The Graylog 5 is still supporting Elasticsearch. However, they already officially announce to no longer support Elasticsearch in the future and recommend migrating to OpenSearch.

I think it is better to remove Elasticsearch and add OpenSearch to the chart. 
If someone wants to use Elasticsearch they can use the previous chart version.

**Comment by KongZ** [comment]
@luizjr please fix chart-testing. :)

```
Error: [trailing-spaces] trailing spaces
```

**Comment by luizjr** [comment]
> @luizjr please fix chart-testing. :)
> 
> ```
> Error: [trailing-spaces] trailing spaces
> ```

I will still do it today!
Stay tuned, let's align this, thanks for the opportunity to help.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -20,11 +24,9 @@ jobs:
...
 
       - name: Install Helm
-        uses: azure/setup-helm@v1
-        with:
-          version: v3.4.0
+        uses: azure/setup-helm@v3
 
       - name: Run chart-releaser
-        uses: helm/chart-releaser-action@v1.2.1
+        uses: helm/chart-releaser-action@v1.5.0
```
Comment (KongZ) [inline .github/workflows/release.yaml:30]: A `chart-releaser-action@v1.5.0` requires to update syntax on `.github/workflows/chart-testing.yaml` line 36

```
------
.github/workflows/chart-testing.yaml:36:14: workflow command "set-output" was deprecated. use `echo "{name}={value}" >> $GITHUB_OUTPUT` instead: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions [deprecated-commands]
   |
36 |         run: |
   |              ^
------
```

**Comment by luizjr** [comment]
I'm getting more from these actions than from the actual helm chart. if anyone can help to be able to publish this update.

**Comment by luizjr** [comment]
I'm getting more from these actions than from the actual helm chart. if anyone can help to be able to publish this update.

File: .github/workflows/chart-testing.yaml [hunk 3]
```diff
@@ -35,8 +35,8 @@ jobs:
         id: list-changed
         run: |
           changed=$(ct list-changed --target-branch main)
-          if [[ -n "$changed" ]]; then
-            echo "::set-output name=changed::true"
+          if [[ -n '$changed' ]]; then
+            echo 'changed=true' >> $GITHUB_OUTPUT
           fi
```
Comment (magikmw) [inline .github/workflows/chart-testing.yaml:40]: This is what is breaking one of the linters.

```suggestion
          if [[ -n "$changed" ]]; then
            echo "changed=true" >> "$GITHUB_OUTPUT"
          fi
```

**Comment by magikmw** [comment]
I can't tell what's wrong exactly with the chart test itself, it seems like the opensearch deployment is failing due to insufficient CPU. I have no experience with github actions, so I'm not sure what to do about it.

**Comment by luizjr** [comment]
> I can't tell what's wrong exactly with the chart test itself, it seems like the opensearch deployment is failing due to insufficient CPU. I have no experience with github actions, so I'm not sure what to do about it.

I observed this better, I'll send another pr to try to solve.

**Comment by KongZ** [comment]
Github Action has little shared resources. You may need to reduce default resources allocation and heap size on OpenSearch chart values.

**Comment by luizjr** [comment]
It seems to me that it worked now. Can you check and tell me if anything is missing?

**Review by KongZ — APPROVED** [review]

**Comment by KongZ** [comment]
@luizjr please set resource back to 
```
      cpu: "200m"
      memory: "512Mi"
```
so the workflow will be pass. Then I can merge and release

**Comment by luizjr** [comment]
I think I'll have to redo the changes, because I opened it with codespaces and it wasn't signed and that's why it doesn't want to merge.

What do you suggest @KongZ ?

**Comment by KongZ** [comment]
@luizjr you can sign-off previous commits by
```
git rebase --signoff <branch>
```
If you could not sign, it is ok. I can merge anyway.

**Comment by luizjr** [comment]
@KongZ Good evening, what do you still need me to do to update the helm chart?

Am I missing something?

**Comment by KongZ** [comment]
The test is running failed because Insufficient resource
```
  Warning  FailedScheduling  8s (x7 over 5m4s)  default-scheduler  0/1 nodes are available: 1 Insufficient cpu.
````

**Comment by KongZ** [comment]
@luizjr I have merged and released https://github.com/KongZ/charts/releases/tag/graylog-2.3.0

**Pull request merged**